be/src/exprs/aggregate_functions.cpp

```diff
@@ -234,20 +234,27 @@ void AggregateFunctions::avg_update(FunctionContext* ctx, const T& src, StringVa
 
 struct PercentileApproxState {
 public:
-    PercentileApproxState() : digest(new TDigest()){
-    }
+    PercentileApproxState() : digest(new TDigest()) {}
+    PercentileApproxState(double compression) : digest(new TDigest(compression)) {}
     ~PercentileApproxState() {
         delete digest;
     }
-    
+
     TDigest *digest = nullptr;
     double targetQuantile = -1.0;
 };
 
 void AggregateFunctions::percentile_approx_init(FunctionContext* ctx, StringVal* dst) {
+    const AnyVal* digest_compression = ctx->get_constant_arg(2);
```

Comment:
You should check the number of arguments, because there is only two arguments if user don't specify compression param

Reply:
If the index exceeds the number of arguments, get_constant_arg() will return a null value. I will check it at line 251.